src/factom-event-emitter.js

```diff
@@ -0,0 +1,274 @@
+const { EventEmitter } = require('events'),
+    Promise = require('bluebird'),
+    { isValidPublicFctAddress } = require('./addresses'),
+    { Transaction } = require('./transaction');
+
+const FACTOM_EVENT = {
+    directoryBlock: 'directoryBlock',
+    factoidBlock: 'factoidBlock',
+    adminBlock: 'adminBlock',
+    entryCreditBlock: 'entryCreditBlock',
+    newEntryChain: 'newEntryChain'
+};
+Object.freeze(FACTOM_EVENT);
+
+/**
+ * Listen for new Factom Events.
+ * @param {FactomCli} cli - FactomCli instance to be used by the FactomEventEmitter instance to poll the blockchain for new blocks.
+ * @param {object} opts - Options to set on the FactomEventEmitter instance
+ * @param {number} [opts.interval=7500] - The interval (ms) at which the FactomEventEmtitter instance should poll the blockchain to check for a new block. Defaults to 7500.
+ * @class
+ */
+class FactomEventEmitter extends EventEmitter {
+    constructor(cli, opts) {
+        super();
+
+        this._cli = cli;
+        this._lastBlockHeightProcessed = 0;
+
+        this.opts = {
+            interval: 7500,
+            ...opts
+        };
+
+        // FCT addresses and chain ids still need to be tracked manually for efficiency
+        this._entryChainSubscriptions = new Set();
+        this._factoidAddressSubscriptions = new Set();
+
+        this.on('removeListener', event => this._removeListener(event));
+        this.on('newListener', event => this._newListener(event));
+    }
+
+    /**
+     * Get active entry chain subscriptions
+     * @returns {Set<string>}
+     */
+    get entryChainSubscriptions() {
+        return this._entryChainSubscriptions;
+    }
+
+    /**
+     * Get active factoid address subscriptions
+     * @returns {Set<string>}
+     */
+    get factoidAddressSubscriptions() {
+        return this._factoidAddressSubscriptions;
+    }
+
+    /**
+     * Determine whether or not polling is currently active.
+     * @returns {boolean}
+     */
+    get isPolling() {
+        return !!this._isPolling;
+    }
+
+    ///////////////////////////////////////////////////////////////
+    //                    MANAGE LISTENERS                      //
+    /////////////////////////////////////////////////////////////
+
+    // This method only starts polling and keeps track of "custom" events such as FCT addresses and chain ids
+    async _newListener(event) {
+        // Should not start polling when listening to non-blockchain events such as 'error'
+        if (this._isBlockchainEvent(event)) {
+            this._startPolling();
+        }
+
+        // Block events are directly handled by the base EventEmitter
+        if (Object.values(FACTOM_EVENT).includes(event)) {
+            return;
+        }
+
+        if (isValidPublicFctAddress(event)) {
+            this._factoidAddressSubscriptions.add(event);
+        } else if (typeof event === 'string' && event.match(/\b[A-Fa-f0-9]{64}\b/)) {
```

Comment:
The base event emitter class does not type check the event, unfortunately. You can pass an object and it will not throw. Not sure how you would want to handle that?

Reply:
Ok, then should we throw (or rather emit) an error at the very begining of this method if the event is not a string?